config/kube.py

```diff
@@ -0,0 +1,359 @@
+
+def _log_debug(m):
+    _log_msg(syslog.LOG_DEBUG, inspect.stack()[1][1], inspect.stack()[1][2], m)
+
+def _do_exit(m):
+    m = "FATAL failure: {}. Exiting...".format(m)
+    _log_msg(syslog.LOG_ERR, inspect.stack()[1][1], inspect.stack()[1][2], m)
+    raise SystemExit(m) 
+
+def _get_db_data(table, key):
+    config_db = ConfigDBConnector()
+    config_db.connect()
+    data = config_db.get_table(table)
+    return data[key] if key in data else None
+
+def _update_kube_server(field, val):
+    config_db = ConfigDBConnector()
+    config_db.connect()
+    table = "KUBERNETES_MASTER"
+    key = "SERVER"
+    db_data = _get_db_data(table, key)
+    def_data = {
+        "IP": "",
+        "insecure": "False",
+        "disable": "False"
+    }
+    for f in def_data:
+        if db_data and f in db_data:
+            if f == field and db_data[f] != val:
+                config_db.mod_entry(table, key, {field:val})
+        else:
+            # Missing field. Set to default or given value
+            v = val if f == field else def_data[f]
+            config_db.mod_entry(table, key, {f:v})
+
+
+def _take_lock():
+    lock_fd = None
+    try:
+        lock_fd = open(LOCK_FILE, "w")
+        fcntl.lockf(lock_fd, fcntl.LOCK_EX | fcntl.LOCK_NB)
+        _log_info("Lock taken {}".format(LOCK_FILE))
+    except IOError as e:
+        lock_fd = None
+        _log_err("Lock {} failed: {}".format(LOCK_FILE, str(e)))
+    return lock_fd
+
+def _download_file(server, insecure):
+    fname = ""
+    if insecure:
+        urllib3.disable_warnings(urllib3.exceptions.InsecureRequestWarning)
+
+    r = requests.get(SERVER_ADMIN_URL.format(server), verify=not insecure)
+    if r.status_code == 200:
+        (h, fname) = tempfile.mkstemp(suffix="_kube_join")
+        os.write(h, r.text)
+        os.close(h)
+    else:
+        _do_exit("Failed to download {}".format(SERVER_ADMIN_URL.format(server)))   
+
+
+    # Ensure the admin.conf has given VIP as server-IP.
+    update_file = "{}.upd".format(fname)
+    cmd = 'sed "s/server:.*:6443/server: https:\/\/{}:6443/" {} > {}'.format(
+            server, fname, update_file)
+    run_command(cmd)
+
+    shutil.copyfile(update_file, KUBE_ADMIN_CONF)
+
+    run_command("rm -f {} {}".format(fname, update_file))
+
+
+def _is_connected(server=""):
+    if (
+            os.path.exists(KUBE_ADMIN_CONF) and
+            os.path.exists(KUBELET_YAML) and
+            os.path.exists(KUBELET_SERVICE)):
+
+        with open(KUBE_ADMIN_CONF, 'r') as s:
+            d = yaml.load(s)
+            d = d['clusters'] if 'clusters' in d else []
+            d = d[0] if len(d) > 0 else {}
+            d = d['cluster'] if 'cluster' in d else {}
+            d = d['server'] if 'server' in d else ""
+            if d:
+                o = urlparse(d)
+                if o.hostname:
+                    return not server or server == o.hostname
+    return False
+                
+
+
+
+def _get_labels():
```

Comment:
Too many empty lines

Reply:
Agreed. @renukamanavalan: Until we have an automated linter in place, I suggest passing all Python code through http://pep8online.com/ and make changes based on the issues it finds (you can ignore the "line too long" errors, unless the line is exceptionally long).

Reply:
Verified against the link provided by Joe. Thanks Joe!